Mathlib/Data/Bool/Basic.lean

```diff
@@ -241,107 +221,66 @@ theorem eq_false_of_not_eq_true' {a : Bool} : !a = true → a = false := by
 
-@[simp]
-theorem xor_assoc : ∀ a b c, xor (xor a b) c = xor a (xor b c) := by decide
+attribute [simp] xor_assoc
 #align bool.bxor_assoc Bool.xor_assoc
 
-theorem xor_left_comm : ∀ a b c, xor a (xor b c) = xor b (xor a c) := by decide
 #align bool.bxor_left_comm Bool.xor_left_comm
-
-@[simp]
-theorem xor_not_left : ∀ a, xor (!a) a = true := by decide
 #align bool.bxor_bnot_left Bool.xor_not_left
-
-@[simp]
-theorem xor_not_right : ∀ a, xor a (!a) = true := by decide
 #align bool.bxor_bnot_right Bool.xor_not_right
 
-@[simp]
-theorem xor_not_not : ∀ a b, xor (!a) (!b) = xor a b := by decide
+attribute [simp] xor_not_not
 #align bool.bxor_bnot_bnot Bool.xor_not_not
 
-@[simp]
-theorem xor_false_left : ∀ a, xor false a = a := by decide
 #align bool.bxor_ff_left Bool.xor_false_left
-
-@[simp]
-theorem xor_false_right : ∀ a, xor a false = a := by decide
 #align bool.bxor_ff_right Bool.xor_false_right
-
-theorem and_xor_distrib_left (a b c : Bool) : (a && xor b c) = xor (a && b) (a && c) := by
-  cases a <;> simp
 #align bool.band_bxor_distrib_left Bool.and_xor_distrib_left
-
-theorem and_xor_distrib_right (a b c : Bool) : (xor a b && c) = xor (a && c) (b && c) := by
-  cases a <;> cases b <;> cases c <;> simp
 #align bool.band_bxor_distrib_right Bool.and_xor_distrib_right
 
 theorem xor_iff_ne : ∀ {x y : Bool}, xor x y = true ↔ x ≠ y := by decide
 #align bool.bxor_iff_ne Bool.xor_iff_ne
 
 /-! ### De Morgan's laws for booleans-/
 
-@[simp]
-theorem not_and : ∀ a b : Bool, (!(a && b)) = (!a || !b) := by decide
+attribute [simp] not_and
 #align bool.bnot_band Bool.not_and
 
-@[simp]
-theorem not_or : ∀ a b : Bool, (!(a || b)) = (!a && !b) := by decide
+attribute [simp] not_or
 #align bool.bnot_bor Bool.not_or
 
-theorem not_inj : ∀ {a b : Bool}, (!a) = !b → a = b := by decide
 #align bool.bnot_inj Bool.not_inj
 
--- Porting note: having to unfold here is not pretty.
--- There is a discussion on zulip about this at
--- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/LinearOrder.20in.20mathlib3.2F4/near/308228493
 instance linearOrder : LinearOrder Bool where
-  le := fun a b ↦ a = false ∨ b = true
-  le_refl := by unfold LE.le; decide
-  le_trans := by unfold LE.le; decide
-  le_antisymm := by unfold LE.le Preorder.toLE; decide
-  le_total := by unfold LE.le Preorder.toLE PartialOrder.toPreorder; decide
-  decidableLE := by unfold LE.le Preorder.toLE PartialOrder.toPreorder; exact inferInstance
-  decidableEq := inferInstance
-  max := or
-  max_def := λ a b => by cases a <;> cases b <;> decide
-  min := and
-  min_def := λ a b => by cases a <;> cases b <;> decide
+  le_refl := Bool.le_refl
+  le_trans _ _ _ := Bool.le_trans
+  le_antisymm _ _ := Bool.le_antisymm
+  le_total := Bool.le_total
+  decidableLE := inferInstance
+  lt_iff_le_not_le _ _ := Bool.lt_iff_le_not_le
 #align bool.linear_order Bool.linearOrder
 
-@[simp] theorem max_eq_or : max = or := rfl
-
-@[simp] theorem min_eq_and : min = and := rfl
```

Comment:
I don't think these lemmas should have been deleted, the versions in Std have different statements.

It would be great if Lean could tell us "this declaration already exists *with the same statement* (suggesting a deletion)" vs "a declaration with the same name already exists (suggesting a rename)"

Reply:
I made them match in std4#329. I don't think this was an intended divergence.